src/coreclr/vm/jitinterface.cpp

```diff
@@ -9875,12 +9871,13 @@ void CEEInfo::getAddressOfPInvokeTarget(CORINFO_METHOD_HANDLE method,
 {
     WRAPPER_NO_CONTRACT;
 
-    void* pIndirection;
     {
         JIT_TO_EE_TRANSITION_LEAF();
```

Comment:
This `JIT_TO_EE_TRANSITION_LEAF` is not correct. `TryGetResolvedNDirectTarget` can throw. JIT/EE interface methods that can throw have to use full `JIT_TO_EE_TRANSITION` to make exception handling work.

I would change this method to:
```
void CEEInfo::getAddressOfPInvokeTarget(CORINFO_METHOD_HANDLE method,
                                        CORINFO_CONST_LOOKUP *pLookup)
{
    CONTRACTL {
        NOTHROW;
        GC_NOTRIGGER;
        MODE_PREEMPTIVE;
    } CONTRACTL_END;

    JIT_TO_EE_TRANSITION();

    MethodDesc* pMD = GetMethod(method);
    _ASSERTE(pMD->IsNDirect());

    NDirectMethodDesc *pNMD = (NDirectMethodDesc*)pMD;

    void* pIndirection;
    if (NDirectMethodDesc::TryGetResolvedNDirectTarget(pNMD, &pIndirection))
    {
        pLookup->accessType = IAT_VALUE;
        pLookup->addr = pIndirection;
    }
    else
    {
        pLookup->accessType = IAT_PVALUE;
        pLookup->addr = (LPVOID)&(pNMD->GetWriteableData()->m_pNDirectTarget);
    }

    EE_TO_JIT_TRANSITION();
}
```
and delete `getAddressOfPInvokeFixup`.

Reply:
Fixed.